src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/AssociatedMetadataTypeTypeDescriptor.cs

```diff
@@ -29,6 +33,7 @@ public AssociatedMetadataTypeTypeDescriptor(ICustomTypeDescriptor parent, Type t
             }
         }
 
+        [RequiresUnreferencedCode("The public parameterless constructor or the 'Default' static field may be trimmed from the Attribute's Type.")]
```

Comment:
Don't we also need an entry on the ILLink.LibraryBuild.xml for every RequiresUnreferencedCode that we add?

Reply:
No, we only need those entries if there are ILLink warnings that aren't being addressed. In this case, we are marking all calls to this API as `RequiresUnreferencedCode` as well, so the trimmer doesn't warn.